### ISSUE TITLE
ColorWheel object has no attribute 'sv_s'

### ISSUE THREAD

**Issue by abstractdonut** [kivy/kivy#6585]
### Versions

* Python: 2.7.15 (3.6.8)
* OS: Ubuntu 18.04 LTS
* Kivy: 1.11.1
* Kivy installation method: pip (pip3)

### Description

I am trying to display a ColorWheel by itself, without the picker. At first the program did not sublcass ColorWheel and the result was the same error. The subclassing involves corrections that I found at https://stackoverflow.com/questions/29199797/kivy-color-wheel. I Googled the error and found https://stackoverflow.com/questions/56097049/unable-to-use-colorwheel-without-loading-kv-attributeerror where the answer indicates that an issue should be submitted, but I don't believe the asker ever followed up.

### Code and Logs

    from kivy.uix.colorpicker import ColorWheel
    from kivy.app import App
    
    class AutonomousColorWheel(ColorWheel):
        def __init__(self, **kwarg):
            super(AutonomousColorWheel, self).__init__(**kwarg)
            self.init_wheel(dt = 0) 
    
        def on__hsv(self, instance, value):
             super(AutonomousColorWheel, self).on__hsv(instance, value)
             print(self.rgba) 
    
    
    class ACWApp(App):
        def build(self):
            return AutonomousColorWheel()
    
    
    if __name__=="__main__":
        ACWApp().run()

Here is the traceback
```
[INFO   ] [Logger      ] Record log in /home/user/.kivy/logs/kivy_19-11-03_11.txt
[INFO   ] [Kivy        ] v1.11.1
[INFO   ] [Kivy        ] Installed at "/usr/lib/python2.7/dist-packages/kivy/__init__.pyc"
[INFO   ] [Python      ] v2.7.15+ (default, Oct  7 2019, 17:39:04) 
[GCC 7.4.0]
[INFO   ] [Python      ] Interpreter at "/usr/bin/python"
[INFO   ] [Logger      ] Purge log fired. Analysing...
[INFO   ] [Logger      ] Purge 7 log files
[INFO   ] [Logger      ] Purge finished!
[WARNING] [Deprecated  ] Python 2 Kivy support has been deprecated. The Kivy release after 1.11.0 will not support Python 2 anymore
[INFO   ] [Factory     ] 184 symbols loaded
[DEBUG  ] [Cache       ] register <kv.image> with limit=None, timeout=60
[DEBUG  ] [Cache       ] register <kv.atlas> with limit=None, timeout=None
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil, img_gif (img_ffpyplayer ignored)
[DEBUG  ] [Cache       ] register <kv.texture> with limit=1000, timeout=60
[DEBUG  ] [Cache       ] register <kv.shader> with limit=1000, timeout=3600
[DEBUG  ] [Cache       ] register <kv.lang> with limit=None, timeout=None
[DEBUG  ] [App         ] Loading kv <./acw.kv>
[DEBUG  ] [App         ] kv <./acw.kv> not found
[DEBUG  ] [Window      ] Ignored <egl_rpi> (import error)
[INFO   ] [Window      ] Provider: sdl2(['window_egl_rpi'] ignored)
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] Backend used <sdl2>
[INFO   ] [GL          ] OpenGL version <3.0 Mesa 19.0.8>
[INFO   ] [GL          ] OpenGL vendor <Intel Open Source Technology Center>
[INFO   ] [GL          ] OpenGL renderer <Mesa DRI Intel(R) UHD Graphics 620 (Kabylake GT2) >
[INFO   ] [GL          ] OpenGL parsed version: 3, 0
[INFO   ] [GL          ] Shading version <1.30>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <32>
[DEBUG  ] [Shader      ] Fragment compiled successfully
[DEBUG  ] [Shader      ] Vertex compiled successfully
[DEBUG  ] [ImageSDL2   ] Load </usr/lib/python2.7/dist-packages/kivy/data/glsl/default.png>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
 Traceback (most recent call last):
   File "main.py", line 21, in <module>
     ACWApp().run()
   File "/usr/lib/python2.7/dist-packages/kivy/app.py", line 829, in run
     root = self.build()
   File "main.py", line 17, in build
     return AutonomousColorWheel()
   File "main.py", line 7, in __init__
     super(AutonomousColorWheel, self).__init__(**kwarg)
   File "/usr/lib/python2.7/dist-packages/kivy/uix/colorpicker.py", line 138, in __init__
     super(ColorWheel, self).__init__(**kwargs)
   File "/usr/lib/python2.7/dist-packages/kivy/uix/widget.py", line 361, in __init__
     rule_children=rule_children)
   File "/usr/lib/python2.7/dist-packages/kivy/uix/widget.py", line 469, in apply_class_lang_rules
     rule_children=rule_children)
   File "/usr/lib/python2.7/dist-packages/kivy/lang/builder.py", line 538, in apply
     rule_children=rule_children)
   File "/usr/lib/python2.7/dist-packages/kivy/lang/builder.py", line 707, in _apply_rule
     e), cause=tb)
 kivy.lang.builder.BuilderException: Parser: File "/usr/lib/python2.7/dist-packages/kivy/data/style.kv", line 1180:
 ...
    1178:
    1179:<ColorWheel>:
 >> 1180:    _origin: self.center
    1181:    _radius: 0.45 * min(self.size)
    1182:
 ...
 AttributeError: 'AutonomousColorWheel' object has no attribute 'sv_s'
   File "/usr/lib/python2.7/dist-packages/kivy/lang/builder.py", line 696, in _apply_rule
     setattr(widget_set, key, value)
   File "kivy/weakproxy.pyx", line 35, in kivy.weakproxy.WeakProxy.__setattr__
   File "kivy/properties.pyx", line 497, in kivy.properties.Property.__set__
   File "kivy/properties.pyx", line 839, in kivy.properties.ListProperty.set
   File "kivy/properties.pyx", line 544, in kivy.properties.Property.set
   File "kivy/properties.pyx", line 599, in kivy.properties.Property.dispatch
   File "kivy/_event.pyx", line 1214, in kivy._event.EventObservers.dispatch
   File "kivy/_event.pyx", line 1120, in kivy._event.EventObservers._dispatch
   File "/usr/lib/python2.7/dist-packages/kivy/uix/colorpicker.py", line 145, in on__origin
     self.init_wheel(None)
   File "/usr/lib/python2.7/dist-packages/kivy/uix/colorpicker.py", line 168, in init_wheel
     self.sv_s[self.sv_idx + r][0],
```

**Comment by KBoehme** [comment]
Any ideas on this issue? Im running into the same thing. 

    Python: 3.9.5
    OS: MacOS
    Kivy: 2.0.0
    Kivy installation method: pip

**Comment by Julian-O** [comment]
As `ColorWheel.__init__()` declares the attribute `self.sv_s` (which appear to be a list of x, y coordinates).

But before that, it initialises its superclass - i.e. `Widget`.

While `Widget` is being initialised, it tries to set the origin which, in turn, implicitly calls `ColorPicker.on__origin()` (note: double underscore!?) which calls `init_wheel()` which relies on `self.sv_s`... which hasn't been initialised.

Potential fixes include:

* `ColorPicker.__init__()` should set the value before calling `super().__init__()`
* `init_wheel()` should not assume `self.sv_s` is set, and set it itself (or recalculate on the fly)

**Comment by Julian-O** [comment]
Oh, and bug confirmed on Windows 11, Kivy 2.2.1.

**Comment by misl6** [comment]
Fixed via #8492
